core/db/migrate/20141217215630_update_product_slug_index.rb

```diff
@@ -0,0 +1,6 @@
+class UpdateProductSlugIndex < ActiveRecord::Migration
+  def change
+    remove_index :spree_products, :slug
+    add_index :spree_products, :slug, unique: true
```

Comment:
A product is paranoid isn't it? Maybe:

```ruby
add_index :spree_product, :slug, unique: true, where: 'deleted_at IS NULL`
```

This way the index doesn't contain deleted data (lower DB size? :D) and if you delete a product then recreate it, you might get an error.